### DISCUSSION TITLE
Job stall and stalledInterval

### DISCUSSION THREAD

**Discussion by aishadeshmukh** [OptimalBits/bull#2156] [Q&A]
Hii

I want to alter the stalledInterval time and check how the job gets stalled. However, I'm unable to find any scenario where the jobs gets stalled. Is there any approach where I can test this? Searching for ways since a long time, but couldn't find any.
Would really appreciate if anyone could help!

Thanks!

**Comment by manast** [comment]
Take also a look at the ```lockDuration``` and ```lockRenewTime``` settings: https://github.com/OptimalBits/bull/blob/develop/REFERENCE.md#queue

If you choose a  ```lockDuration``` of say 1 second, but the ```lockRenewTime``` is 5 seconds, and a ```stalledInterval``` of 1 second, then you will certainly get stalled jobs if they run for more than one second.

**Reply by aishadeshmukh** [reply]
Thanks for the response!

I want to implement a logic where a job can take long time, (due to which it will go into stalled state) but shouldn't get stalled.

I used below functions to get the job stalled

`  const hash = crypto.createHash('sha256');
    const end = Date.now() + 5000;
    while (Date.now() < end) {
      hash.update(this.randomString())
    }  `
 
    And then used below to update the progress of the job
    
    `let interval = setInterval(() => {
      console.log('COUNTER---> ', counter)
      bullJobWebhookPayload.progress(counter)
      counter = counter + 1;
    }, 2000)`


and these are my config settings
`settings: {
        maxStalledCount: 0,
        stalledInterval: 10000,
      },`

  I'm clearing the interval before the main function return success
    Now, what happens here is, the job gets stalled after a certain amount of time and then I get job error of Missing Lock.
    I tried changing the while loops (using only one or both) but that still gives me the same issue. The setInterval() works only when I apply a sleep timer of 3000 in my main function (await), else does not work.
    **Requirement**
    I want to update the progress of the jobs, so that if it's a longer job, bull will know that it's in progress and won't stall the job.
    Even if the setInterval works(by adding sleep in function), it works until the stalledInterval time is reached, after which it stalls the jobs and then continues until the job error of Missing Lock.
    
 
 Could you please guide me on how to achieve this? Thanks in advance!

**Reply by manast** [reply]
Could you paste some code that can be run as is and that demonstrates the case?
Regarding the missing lock, this usually happens when the job stalls but then the worker that actually stalled the job finally completes and tries to move the job to "completed" status, but since the job has stalled and has been moved back to the "wait" list, the worker do not have a lock on the job anymore and fails.

**Reply by manast** [reply]
If it is the case above then the error can be ignored and just kept as a log for reference.
